### ISSUE TITLE
Getting skip components

### ISSUE THREAD

**Issue by blefloch** [latex3/latex3#497]
It seems we have no function to extract the main/stretch/shrink components of a skip or the degrees of infinite stretch or shrink components.  I have no idea how to name these functions.  Perhaps

    \skip_main:n % \tex_dimexpr:D
    \skip_stretch:n % \tex_gluestretch:D
    \skip_shrink:n % \tex_glueshrink:D
    \skip_stretch_order:n % \tex_gluestretchorder:D
    \skip_shrink_order:n % \tex_glueshrinkorder:D

We should perhaps have wrappers around `\tex_mutoglue:D` and `\tex_gluetomu:D`.

Perhaps we could have a function to convert a skip to a tuple of three floating point numbers, with orders of infinities represented by some suitable large exponent.

**Comment by eg9** [comment]
> On 3 Nov 2018, at 21:43 , Bruno Le Floch <notifications@github.com> wrote:
> 
> It seems we have no function to extract the main/stretch/shrink components of a skip or the degrees of infinite stretch or shrink components. I have no idea how to name these functions. Perhaps
> 
> \skip_main:n % \tex_dimexpr:D
> \skip_stretch:n % \tex_gluestretch:D
> \skip_shrink:n % \tex_glueshrink:D
> \skip_stretch_order:n % \tex_gluestretchorder:D
> \skip_shrink_order:n % \tex_glueshrinkorder:D
> 
> We should perhaps have wrappers around \tex_mutoglue:D and \tex_gluetomu:D.
> 
> Perhaps we could have a function to convert a skip to a tuple of three floating point numbers, with orders of infinities represented by some suitable large exponent.

Not sure about the last proposal. Surely these functions
will need to find their place, sooner or later.

Maybe \skip_natural:n for the natural width.

I’m not sure about the usefulness of \tex_mutoglue:D and
\tex_gluetomu:D, but they’re there, so I guess an interface
should exist.

Ciao
Enrico

**Comment by FrankMittelbach** [comment]
I disagree. An interface to these components is necessary in layout (give or take the name and that it took us 5 years of doing nothing about it yet). But that means we might as well keep this pr option and label it longterm.

**Comment by josephwright** [comment]
I'm happy leaving as a feature request without other labelling: if we know it's been looked at and is focussed enough, it's good enough for me.

**Comment by FrankMittelbach** [comment]
Seeing that you want to clean out the main issue log (which dies make sense) I wonder if it would be possible to simply add a project for such "long-term might get looked at one day" topics, and put them in there in a closed state so that they are easy to revisit via the project instead of having to look through all closed issues (most of which have been closed for good reason forever).

**Comment by car222222** [comment]
Such ideas (from @FrankMittelbach) seem to be along the right lines.
